cmd/snap/cmd_snap_op.go

```diff
@@ -962,14 +975,41 @@ func (x cmdSwitch) Execute(args []string) error {
 	if err := x.setChannelFromCommandline(); err != nil {
 		return err
 	}
-	if x.Channel == "" {
-		return fmt.Errorf("missing --channel=<channel-name> parameter")
-	}
 
 	name := string(x.Positional.Snap)
 	channel := string(x.Channel)
+
+	var msg string
+	// some duplication between this and the two other switch-summarisers...
+	// in this one, we have three boolean things to check, meaning 2³=8 possibilities
+	// of which 3 are errors (which is why we look at this before running it)
+	switch {
+	case x.Cohort != "" && x.LeaveCohort:
+		// this one counts as two (no channel filter)
+		return fmt.Errorf(i18n.G("cannot request both --cohort and --leave-cohort"))
+	case x.Cohort != "" && !x.LeaveCohort && channel == "":
+		// TRANSLATORS: the first %q will be the (quoted) snap name, the second an ellipted cohort string
+		msg = fmt.Sprintf(i18n.G("%q switched to the %q cohort\n"), name, strutil.ElliptRight(x.Cohort, 10))
+	case x.Cohort != "" && !x.LeaveCohort && channel != "":
+		// TRANSLATORS: the first %q will be the (quoted) snap name, the second a channel, the third an ellipted cohort string
+		msg = fmt.Sprintf(i18n.G("%q switched to the %q channel in the %q cohort\n"), name, channel, strutil.ElliptRight(x.Cohort, 10))
+	case x.Cohort == "" && !x.LeaveCohort && channel != "":
+		// TRANSLATORS: the first %q will be the (quoted) snap name, the second a channel
+		msg = fmt.Sprintf(i18n.G("%q switched to the %q channel\n"), name, channel)
+	case x.Cohort == "" && x.LeaveCohort && channel != "":
+		// TRANSLATORS: the first %q will be the (quoted) snap name, the second a channel
+		msg = fmt.Sprintf(i18n.G("%q left the cohort, and switched to the %q channel"), name, channel)
+	case x.Cohort == "" && x.LeaveCohort && channel == "":
+		// TRANSLATORS: %q will be the (quoted) snap name
+		msg = fmt.Sprintf(i18n.G("%q left the cohort"), name)
+	case x.Cohort == "" && !x.LeaveCohort && channel == "":
+		return fmt.Errorf(i18n.G("nothing to switch (specify --cohort=... or --leave-cohort, and/or --channel=...)"))
+	} // and that's the 8 \o/
```

Comment:
I'm not sure if that helps, but with repeated checks like these I'd consider two helper bools:
```
switchCohort := (x.Cohort != "")
switchChannel := (channel != "")
```
which would make all the conditons slightly more compact/readable. 

Feel free to treat this as completely optional, probably a matter of personal preference.